tests/run/main-annotation-homemade-annot-1.scala

```diff
@@ -0,0 +1,46 @@
+import scala.concurrent._
+import scala.annotation.*
+import scala.collection.mutable
+import ExecutionContext.Implicits.global
+import duration._
+import util.CommandLineParser.FromString
+
+@mainAwait def get(wait: Int): Future[Int] = Future{
+  Thread.sleep(1000 * wait)
+  42
+}
+
+@mainAwait def getMany(wait: Int*): Future[Int] = Future{
+  Thread.sleep(1000 * wait.sum)
+  wait.length
+}
+
+object Test:
+  def callMain(cls: String, args: Array[String]): Unit =
+    val clazz = Class.forName(cls)
+    val method = clazz.getMethod("main", classOf[Array[String]])
+    method.invoke(null, args)
+
+  def main(args: Array[String]): Unit =
+    println(Await.result(get(1), Duration(2, SECONDS)))
+    callMain("get", Array("1"))
+    callMain("getMany", Array("1"))
+    callMain("getMany", Array("0", "1"))
+end Test
+
+@experimental
+class mainAwait(timeout: Int = 2) extends MainAnnotation[FromString, Future[Any]]:
+  import MainAnnotation.*
+
+  // This is a toy example, it only works with positional args
+  def command(info: Info, args: Seq[String]): Option[Seq[String]] = Some(args)
```

Comment:
I would propose to define `command` in the `MainAnnotations` with `Some(args)` implementation. People who need to change this logic can override it here.

Reply:
I like the idea but I'm not sure how many implementations would use this default. I would keep it abstract and see how users are using it before stabilizing this feature. I also expect other practical tweaks based on feedback from users.

Reply:
Ok, you're right. We can always provide the default later but changing an existing default would be a breaking change.